### PR TITLE
feat: clone tenant

### PR DESCRIPTION
**Pull request by benPearce1**
depends on: https://github.com/OctopusDeploy/go-octopusdeploy/pull/176

interactive:
image

automated
image

result:
image

### CODE REVIEW

File: pkg/cmd/tenant/clone/clone.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+			opts := NewCloneOptions(cloneFlags, cmd.NewDependencies(f, c))
+
+			return cloneRun(opts)
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&cloneFlags.Name.Value, cloneFlags.Name.Name, "n", "", "Name or Id of the new tenant")
+	flags.StringVarP(&cloneFlags.Description.Value, cloneFlags.Description.Name, "d", "", "Description of the new tenant")
+	flags.StringVar(&cloneFlags.SourceTenant.Value, cloneFlags.SourceTenant.Name, "", "Name or Id of the source tenant")
```
[the remark this answers is not in the document]
Comment (benPearce1) [reply pkg/cmd/tenant/clone/clone.go:80]: We have a number of flags that are name/id/slug and have built lookup functions in the client to support finding a single entity by any value.
Do we want behaviour consistency, name only everywhere, or hybrid?

[the remark this answers is not in the document]
Comment (jbristowe) [reply pkg/cmd/tenant/clone/clone.go:80]: I vote for `name-only` everywhere (change all the things).

[the remark this answers is not in the document]
Comment (jbristowe) [reply pkg/cmd/tenant/clone/clone.go:80]: Let's start here with `name-only` and then we'll move everything away from IDs elsewhere afterward.

**Review by domenicsim1 — COMMENTED** [review]
Looks good, will ✅  once ID is dropped from descriptions on the clone command.

**Review by jbristowe — APPROVED** [review]

**Pull request merged**